src/snippetor-Snippetor-Templates.ts

```diff
@@ -206,3 +206,479 @@ for (let i = 0; i <= max; i++) {
+        no-repeat;
+    margin-right: 4px !important;
+}
+
+<% } %>
+
+
+<%
+
+defineColors('default', it.cfg.default);
+
+%>
+
+/** Default Folder */
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder.mod-root
+    > .nav-folder-children
+    > .nav-folder
+    > .nav-folder-title,
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder.mod-root
+    > .nav-folder-children
+    > .nav-folder
+    > .nav-folder-title
+    + .nav-folder-children {
+    background-color: var(--<%~ root %>-folder-default);
+    color: var(--<%~ root %>-folder-text-default);
+}
+
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder.mod-root
+    > .nav-folder-children
+    > .nav-folder
+    > .nav-folder-title
+    .nav-folder-title-content,
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder.mod-root
+    > .nav-folder-children
+    > .nav-folder
+    > .nav-folder-title
+    + .nav-folder-children .nav-folder-title-content, 
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder.mod-root
+    > .nav-folder-children
+    > .nav-folder
+    > .nav-folder-title
+    + .nav-folder-children .nav-file-title-content {
+    color: var(--<%~ root %>-folder-text-default);
+}
+
+/** Set the background color of folder and file icons */
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder-title + .nav-folder-children .nav-file-title::before {
+    background-color: var(--<%~ root %>-folder-text-default) !important;
+}
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder-title .nav-folder-title-content::before {
+    background-color: var(--<%~ root %>-folder-text-default) !important;
+}
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder-title + .nav-folder-children .nav-folder-collapse-indicator.collapse-icon > svg.right-triangle {
+    background-color: var(--<%~ root %>-folder-text-default);
+}
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder.mod-root > .nav-folder-children > .nav-folder > .nav-folder-title {
+    border-radius: var(--<%~ root %>-folder-radius) var(--<%~ root %>-folder-radius) 0px 0px;
+}
+
+/** Give the top level folder name a bottom border */
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder.mod-root
+    > .nav-folder-children
+    > .nav-folder:not(.is-collapsed)
+    > .nav-folder-title {
+    border-bottom: var(--<%~ root %>-folder-text-default) 1px solid;
+}
+.workspace-leaf-content[data-type="file-explorer"] .nav-folder.mod-root
+    > .nav-folder-children
+    > .nav-folder
+    > .nav-folder-children {
+    padding-left: 0px;
+    color: var(--<%~ root %>-folder-text-default) ;
+}
+
+<%
+
+for (let i = 0; i<it.cfg.folders.length; i++) {
+    const folder = it.cfg.folders[i];
+    defineColors(i, folder);
+    folderSettings(i, folder);
+}
+
+%>
+
```

Comment:
The one thing I will suggest: With the checkbox settings, I was asked to _only style the specified values_, and nothing else. So the snippets I generate are subsequently more verbose, because I don't use general settings that would apply to all task lists, everything is glommed onto individual items. 

Given you're overlaying themes that will also muck with the nav settings, you may end up being asked to only set the background/icon on the selected folders, and to leave the rest alone.